src/pyFAI/gui/tasks/ExperimentTask.py

```diff
@@ -170,11 +170,15 @@ def _updateModel(self, model):
         self._wavelength.setModel(settings.wavelength())
         self._energy.setModel(settings.wavelength())
 
+        self.__parallaxChangedUpdateModel()  # Initialize & wire
+        self._detectorParallax.clicked.connect(self.__parallaxChangedUpdateModel)
```

Comment:
`clicked` is only called upon user interaction and not when the state is changed from the API.
Is it really what you want?
It's usually best to use `toggled` which is emitted for any state change.

Reply:
I finally implemented QToolButton which is more compact...